### PR TITLE
Re-write global header guidance

### LINKED ISSUE #1607: Add page and guidance for HPE global header
To help users migrate to the global header it would be nice to document and offer guidance from the Design System site.

### PR DESCRIPTION
**Pull request by MikeKingdom**
#### What does this PR do?
Re-write the Global Header template and examples to show how to use the HPE Global Header

#### Where should the reviewer start?
global-header.mdx

#### What are the relevant issues?
Fixes #1607

### CODE REVIEW

File: aries-site/src/pages/templates/global-header.mdx [hunk 1]
```diff
@@ -1,73 +1,127 @@
 import {
-  AppIdentityExample,
   GlobalHeaderExample,
-  HeaderNavItems,
-  SidebarLayer,
 } from '../../examples';
 import { Example } from '../../layouts';
-import { Box } from 'grommet';
+import { Box, Image } from 'grommet';
+import { AppsRounded, Language, Search, Shop } from 'grommet-icons';
```
Comment (halocline) [inline aries-site/src/pages/templates/global-header.mdx:6]:
```suggestion
import { Box, , Button, Image } from 'grommet';
import { AppsRounded, Language, Search, Shop } from 'grommet-icons';

The Global Header, also known as the HPE Common Header and Footer Web Service, is a standardized header to be used across all HPE applications. The header is provided as a service and injected into your application at run-time.

<Button label="View Implementation Details" href="https://h50007.www5.hpe.com/hfws-test-pages/common-hfws-test-page.php" target="_blank" rel="noreferrer  noopener" secondary />

<Button label="Development Team Contact" href="mailto:hpecom-cross-platform-development@hpe.com?subject=Common%20Header-Footer%20Service" secondary />
```

File: aries-site/src/pages/templates/global-header.mdx [hunk 2]
```diff
@@ -1,73 +1,127 @@
...
 ## Guidelines
 
-The Global Header is a standardized header to be used across applications.
+The Global Header is a standardized header to be used across applications. HPE provides an implementation
+that you can reference from your application. You can choose various features to enable like country selector,
+shopping cart and HPE passport login, etc. You can also enable or disable the page footer with privacy statement
+link, terms of use, ad choices and cookies, and *Do Not Sell My Information* links.
+
+Each site or application will have different needs. Some header options may not make sense for some
+sites. The following sections will help you decide which options to include.
```
Comment (halocline) [inline aries-site/src/pages/templates/global-header.mdx:26]:
```suggestion
The Global Header is a standardized header to be used across applications. HPE [provides an implementation](https://h50007.www5.hpe.com/hfws-test-pages/common-hfws-test-page.php)
to be referenced from your application. 

Both the header and footer are customizable, allowing consumers to choose various features to enable such as country selector,
shopping cart, HPE passport login, privacy statement links, theme mode, and more.
Each site or application will have different needs. Some header options may not make sense for some
sites. The following sections will help you decide which options to include.
```

File: aries-site/src/pages/templates/global-header.mdx [hunk 3]
```diff
@@ -1,73 +1,127 @@
...
-  horizontalLayout
-  height="none"
->
-  <HeaderNavItems />
-</Example>
+### Search
+
+The *Search* icon allows the user to search hpe.com. 
+
+<Search />
```
Comment (halocline) [inline aries-site/src/pages/templates/global-header.mdx:69]: Present inline?

```suggestion
<Search /> The *Search* icon allows the user to search hpe.com. 
```

File: aries-site/src/pages/templates/global-header.mdx [hunk 4]
```diff
@@ -1,73 +1,127 @@
...
+### Cube Menu
+
+The *Cube Menu* allows access to HPE services available to the user such as GreenLake and
+other Cloud console applications as well as general HPE links for HPE content like
+developer.hpe.com, education.hpe.com, etc. This menu should be enabled to allow access
+to other HPE sites and applications.
+
+<AppsRounded />
+
+## What makes up the Global footer
```
Comment (halocline) [inline aries-site/src/pages/templates/global-header.mdx:102]:
```suggestion
## What makes up the Global footer?
```

File: aries-site/src/pages/templates/global-header.mdx [hunk 5]
```diff
@@ -1,73 +1,127 @@
...
+
+### Cube Menu
+
+The *Cube Menu* allows access to HPE services available to the user such as GreenLake and
+other Cloud console applications as well as general HPE links for HPE content like
+developer.hpe.com, education.hpe.com, etc. This menu should be enabled to allow access
+to other HPE sites and applications.
+
+<AppsRounded />
+
```
Comment (halocline) [inline aries-site/src/pages/templates/global-header.mdx:101]: Include Color Scheme option?

File: aries-site/src/pages/templates/global-header.mdx [hunk 6]
```diff
@@ -1,73 +1,127 @@
...
+The *Ad Choices and Cookies* link goes to the section of the *Privacy Statement* that explains how
+cookies are used as well as how you can opt-out of targetted advertisements and analytics data
+such as Google Analytics and Adobe Analytics.
+
+
+### Do Not Sell My Personal Information Link
+
+The *Do Not Sell My Personal Information* link goes to information on how to opt-out of any sale
+of personal collected by automatic data collection tools.
```
Comment (halocline) [inline aries-site/src/pages/templates/global-header.mdx:127]: Maybe include links again for convenience? Soft opinion.

```suggestion
of personal collected by automatic data collection tools.


<Box direction="row" gap="medium">
    <Button label="View Implementation Details" href="https://h50007.www5.hpe.com/hfws-test-pages/common-hfws-test-page.php" target="_blank" rel="noreferrer  noopener" secondary />
    <Button label="Development Team Contact" href="mailto:hpecom-cross-platform-development@hpe.com?subject=Common%20Header-Footer%20Service" secondary />
</Box>
```

File: aries-site/src/examples/templates/global-header/components/MockGlobalHeader.js [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+      <Image src="/templateImages/globalheader.png" />
+      { children }
+      <Image src="/templateImages/globalheader-footer.png" />
+    </Box>
+  );
+};
+
+MockGlobalHeader.propTypes = {
+  children: PropTypes.node,
+};
```
Comment (taysea) [inline aries-site/src/examples/templates/global-header/components/MockGlobalHeader.js:25]: Do we need new line>

File: aries-site/src/pages/templates/global-header.mdx (hunk 1 above)
Comment (MikeKingdom) [reply aries-site/src/pages/templates/global-header.mdx:6]: Sounds great.

File: aries-site/src/pages/templates/global-header.mdx (hunk 2 above)
Comment (MikeKingdom) [reply aries-site/src/pages/templates/global-header.mdx:26]: Sounds great

File: aries-site/src/pages/templates/global-header.mdx (hunk 3 above)
Comment (MikeKingdom) [reply aries-site/src/pages/templates/global-header.mdx:69]: Done

File: aries-site/src/pages/templates/global-header.mdx (hunk 4 above)
Comment (MikeKingdom) [reply aries-site/src/pages/templates/global-header.mdx:102]: Done

File: aries-site/src/pages/templates/global-header.mdx (hunk 6 above)
Comment (MikeKingdom) [reply aries-site/src/pages/templates/global-header.mdx:127]: done

File: aries-site/src/pages/templates/global-header.mdx (hunk 5 above)
Comment (MikeKingdom) [reply aries-site/src/pages/templates/global-header.mdx:101]: I wasn't sure what to do with that option. I think I'll check if there's some guidance we can provide for this choice.

Comment (halocline) [reply aries-site/src/pages/templates/global-header.mdx:105]: This is the most documentation I've seen and it doesn't say much.
[common-hfws-quick-js-guide-1.0.0 (3).pdf](https://github.com/grommet/hpe-design-system/files/6593348/common-hfws-quick-js-guide-1.0.0.3.pdf)

What about:
"Color Scheme allows implementors to choose either a light or dark variant of the header/footer. The color scheme you choose should be harmonious with the primary background of your application's header and avoid stark contrast which interrupts a user's visual flow. If your application supports light and dark modes, consider setting the `color_scheme` parameter with a value of either `light` or `dark` in alignment with the user's preference."

To the last point, in our implementation of Global Header, we could reference the localStorage variable `darkMode` --> there is a `useDarkMode()` hook in utils.

**Review by halocline — COMMENTED** [review]
Provided suggestion for Color Scheme guidance.

File: aries-site/src/pages/templates/global-header.mdx (hunk 5 above)
Comment (MikeKingdom) [reply aries-site/src/pages/templates/global-header.mdx:105]: Thanks for the wording. I think that works. I like the useDarkMode hook but I don't think we should recommend using an undocumented grommet API.

Comment (halocline) [reply aries-site/src/pages/templates/global-header.mdx:112]: > I don't think we should recommend using an undocumented grommet API.

Agree. Was meaning that on the Design System site we should change the Global Header color_scheme to match the themeMode someone has chosen for the site.

Comment (halocline) [reply aries-site/src/pages/templates/global-header.mdx:112]: That would be an enhancement outside of this PR, it just happened to be an aside thought I had as I was writing the guidance for the page.

**Review by halocline — APPROVED** [review]
Good to go by me. Thanks @MikeKingdom .

@taysea and I chatted about the one Applitools snapshot being off and will keep eyes on it. Go ahead and merge.

**Comment by MikeKingdom** [comment]
> Good to go by me. Thanks @MikeKingdom .
> 
> @taysea and I chatted about the one Applitools snapshot being off and will keep eyes on it. Go ahead and merge.

I'll check with Lozzi too and see where he's going so we're not just chasing our tail.

**Review by taysea — APPROVED** [review]
I am comfortable with these changes, but happy to wait for Chris's feedback/any copy feedback that the marketing team might have.

I would say this content is more accurate than what we have now though, so wondering if we want to merge then revise once Chris/others provide feedback?

**Pull request merged**